src/Core/Components/NavMenu/FluentNavLink.razor.cs

```diff
@@ -33,4 +40,213 @@ public FluentNavLink()
     {
         _renderContent = RenderContent;
     }
+
+    protected override void OnInitialized()
+    {
+        // We'll consider re-rendering on each location change
+        NavigationManager.LocationChanged += OnLocationChanged;
+    }
+
+    private void OnLocationChanged(object? sender, LocationChangedEventArgs args)
+    {
+        // We could just re-render always, but for this component we know the
+        // only relevant state change is to the _isActive property.
+        var shouldBeActiveNow = ShouldMatch(args.Location);
+        if (shouldBeActiveNow != _isActive)
+        {
+            _isActive = shouldBeActiveNow;
+            UpdateCssClass();
+            StateHasChanged();
+        }
+    }
+
+    /// <inheritdoc />
+    protected override void OnParametersSet()
+    {
```

Comment:
Is it possible to move this code to the `OnInitialized` or `OnAfterRendered` methods? To avoid to execute this code multiple times, for each parameter changes.

Reply:
This is how it is done in the aspnetcore code as well. All parameter changes can lead to active state changes so must be evaluated every time